src/store/message/store.js

```diff
@@ -81,30 +89,36 @@ class MessageStore extends CoreStore {
     addClearMessagesListener(cb) {
         this.addListener(CLEAR, cb);
     }
+
     /**
    * Remove a listener on the global change on the search store.
    * @param {function} cb - The callback to called when a message is pushed.
    */
     removeClearMessagesListener(cb) {
         this.removeListener(CLEAR, cb);
     }
+
+    /**
+     * The store register itself on the dispatcher.
+     *
+     * @memberof MessageStore
+     */
     registerDispatcher() {
-        let currentStore = this;
-        this.dispatch = AppDispatcher.register(function (transferInfo) {
+        this.dispatch = AppDispatcher.register((transferInfo) => {
             let rawData = transferInfo.action.data;
             let type = transferInfo.action.type;
 
             switch (type) {
                 case 'push':
                     if (rawData.message) {
-                        currentStore.pushMessage(rawData.message);
+                        this.pushMessage(rawData.message);
                     }
                     break;
                 case 'clear':
-                    if (rawData.messages) {
```

Comment:
isn't the if statement a fail-safe against a generic `clear` generic action type ?

Reply:
Yeah you are right.
I'm thinking now that we should add an identifier on that store : there are methods to use the store, and I don't think there should any dispatch on that.